// @aws-sdk/client-sns/dist/es/commands/CreateTopicCommand.js
import { __extends } from "tslib";
import { CreateTopicInput, CreateTopicResponse } from "../models/models_0";
import { deserializeAws_queryCreateTopicCommand, serializeAws_queryCreateTopicCommand } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
var CreateTopicCommand = /** @class */ (function (_super) {
    __extends(CreateTopicCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function CreateTopicCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    CreateTopicCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SNSClient";
        var commandName = "CreateTopicCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateTopicInput.filterSensitiveLog,
            outputFilterSensitiveLog: CreateTopicResponse.filterSensitiveLog,
        };
        if (typeof logger.info === "function") {
            logger.info({
                clientName: clientName,
                commandName: commandName,
            });
        }
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateTopicCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreateTopicCommand(input, context);
    };
    CreateTopicCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreateTopicCommand(output, context);
    };
    return CreateTopicCommand;
}($Command));
export { CreateTopicCommand };
