// @aws-sdk/client-sns/dist/es/commands/CreatePlatformApplicationCommand.js
import { __extends } from "tslib";
import { CreatePlatformApplicationInput, CreatePlatformApplicationResponse } from "../models/models_0";
import { deserializeAws_queryCreatePlatformApplicationCommand, serializeAws_queryCreatePlatformApplicationCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
var CreatePlatformApplicationCommand = /** @class */ (function (_super) {
    __extends(CreatePlatformApplicationCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function CreatePlatformApplicationCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    CreatePlatformApplicationCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SNSClient";
        var commandName = "CreatePlatformApplicationCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreatePlatformApplicationInput.filterSensitiveLog,
            outputFilterSensitiveLog: CreatePlatformApplicationResponse.filterSensitiveLog,
        };
        if (typeof logger.info === "function") {
            logger.info({
                clientName: clientName,
                commandName: commandName,
            });
        }
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreatePlatformApplicationCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryCreatePlatformApplicationCommand(input, context);
    };
    CreatePlatformApplicationCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryCreatePlatformApplicationCommand(output, context);
    };
    return CreatePlatformApplicationCommand;
}($Command));
export { CreatePlatformApplicationCommand };
