// @aws-sdk/client-sns/dist/es/commands/GetEndpointAttributesCommand.js
import { __extends } from "tslib";
import { GetEndpointAttributesInput, GetEndpointAttributesResponse } from "../models/models_0";
import { deserializeAws_queryGetEndpointAttributesCommand, serializeAws_queryGetEndpointAttributesCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
var GetEndpointAttributesCommand = /** @class */ (function (_super) {
    __extends(GetEndpointAttributesCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function GetEndpointAttributesCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    GetEndpointAttributesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SNSClient";
        var commandName = "GetEndpointAttributesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetEndpointAttributesInput.filterSensitiveLog,
            outputFilterSensitiveLog: GetEndpointAttributesResponse.filterSensitiveLog,
        };
        if (typeof logger.info === "function") {
            logger.info({
                clientName: clientName,
                commandName: commandName,
            });
        }
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetEndpointAttributesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetEndpointAttributesCommand(input, context);
    };
    GetEndpointAttributesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetEndpointAttributesCommand(output, context);
    };
    return GetEndpointAttributesCommand;
}($Command));
export { GetEndpointAttributesCommand };
