// @aws-sdk/client-sns/dist/es/commands/GetPlatformApplicationAttributesCommand.js
import { __extends } from "tslib";
import { GetPlatformApplicationAttributesInput, GetPlatformApplicationAttributesResponse } from "../models/models_0";
import { deserializeAws_queryGetPlatformApplicationAttributesCommand, serializeAws_queryGetPlatformApplicationAttributesCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
var GetPlatformApplicationAttributesCommand = /** @class */ (function (_super) {
    __extends(GetPlatformApplicationAttributesCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function GetPlatformApplicationAttributesCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    GetPlatformApplicationAttributesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SNSClient";
        var commandName = "GetPlatformApplicationAttributesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetPlatformApplicationAttributesInput.filterSensitiveLog,
            outputFilterSensitiveLog: GetPlatformApplicationAttributesResponse.filterSensitiveLog,
        };
        if (typeof logger.info === "function") {
            logger.info({
                clientName: clientName,
                commandName: commandName,
            });
        }
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetPlatformApplicationAttributesCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetPlatformApplicationAttributesCommand(input, context);
    };
    GetPlatformApplicationAttributesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetPlatformApplicationAttributesCommand(output, context);
    };
    return GetPlatformApplicationAttributesCommand;
}($Command));
export { GetPlatformApplicationAttributesCommand };
