// @aws-sdk/client-sns/dist/es/commands/ListSubscriptionsByTopicCommand.js
import { __extends } from "tslib";
import { ListSubscriptionsByTopicInput, ListSubscriptionsByTopicResponse } from "../models/models_0";
import { deserializeAws_queryListSubscriptionsByTopicCommand, serializeAws_queryListSubscriptionsByTopicCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
var ListSubscriptionsByTopicCommand = /** @class */ (function (_super) {
    __extends(ListSubscriptionsByTopicCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function ListSubscriptionsByTopicCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    ListSubscriptionsByTopicCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SNSClient";
        var commandName = "ListSubscriptionsByTopicCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListSubscriptionsByTopicInput.filterSensitiveLog,
            outputFilterSensitiveLog: ListSubscriptionsByTopicResponse.filterSensitiveLog,
        };
        if (typeof logger.info === "function") {
            logger.info({
                clientName: clientName,
                commandName: commandName,
            });
        }
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListSubscriptionsByTopicCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListSubscriptionsByTopicCommand(input, context);
    };
    ListSubscriptionsByTopicCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListSubscriptionsByTopicCommand(output, context);
    };
    return ListSubscriptionsByTopicCommand;
}($Command));
export { ListSubscriptionsByTopicCommand };
