// @aws-sdk/client-sns/dist/es/commands/ListTopicsCommand.js
import { __extends } from "tslib";
import { ListTopicsInput, ListTopicsResponse } from "../models/models_0";
import { deserializeAws_queryListTopicsCommand, serializeAws_queryListTopicsCommand } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
var ListTopicsCommand = /** @class */ (function (_super) {
    __extends(ListTopicsCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function ListTopicsCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    ListTopicsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SNSClient";
        var commandName = "ListTopicsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListTopicsInput.filterSensitiveLog,
            outputFilterSensitiveLog: ListTopicsResponse.filterSensitiveLog,
        };
        if (typeof logger.info === "function") {
            logger.info({
                clientName: clientName,
                commandName: commandName,
            });
        }
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListTopicsCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListTopicsCommand(input, context);
    };
    ListTopicsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListTopicsCommand(output, context);
    };
    return ListTopicsCommand;
}($Command));
export { ListTopicsCommand };
