// @aws-sdk/client-sns/dist/es/commands/ListPhoneNumbersOptedOutCommand.js
import { __extends } from "tslib";
import { ListPhoneNumbersOptedOutInput, ListPhoneNumbersOptedOutResponse } from "../models/models_0";
import { deserializeAws_queryListPhoneNumbersOptedOutCommand, serializeAws_queryListPhoneNumbersOptedOutCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
var ListPhoneNumbersOptedOutCommand = /** @class */ (function (_super) {
    __extends(ListPhoneNumbersOptedOutCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function ListPhoneNumbersOptedOutCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    ListPhoneNumbersOptedOutCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SNSClient";
        var commandName = "ListPhoneNumbersOptedOutCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListPhoneNumbersOptedOutInput.filterSensitiveLog,
            outputFilterSensitiveLog: ListPhoneNumbersOptedOutResponse.filterSensitiveLog,
        };
        if (typeof logger.info === "function") {
            logger.info({
                clientName: clientName,
                commandName: commandName,
            });
        }
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListPhoneNumbersOptedOutCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryListPhoneNumbersOptedOutCommand(input, context);
    };
    ListPhoneNumbersOptedOutCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryListPhoneNumbersOptedOutCommand(output, context);
    };
    return ListPhoneNumbersOptedOutCommand;
}($Command));
export { ListPhoneNumbersOptedOutCommand };
