// @aws-sdk/client-sns/dist/es/commands/OptInPhoneNumberCommand.js
import { __extends } from "tslib";
import { OptInPhoneNumberInput, OptInPhoneNumberResponse } from "../models/models_0";
import { deserializeAws_queryOptInPhoneNumberCommand, serializeAws_queryOptInPhoneNumberCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
var OptInPhoneNumberCommand = /** @class */ (function (_super) {
    __extends(OptInPhoneNumberCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function OptInPhoneNumberCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    OptInPhoneNumberCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SNSClient";
        var commandName = "OptInPhoneNumberCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: OptInPhoneNumberInput.filterSensitiveLog,
            outputFilterSensitiveLog: OptInPhoneNumberResponse.filterSensitiveLog,
        };
        if (typeof logger.info === "function") {
            logger.info({
                clientName: clientName,
                commandName: commandName,
            });
        }
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    OptInPhoneNumberCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryOptInPhoneNumberCommand(input, context);
    };
    OptInPhoneNumberCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryOptInPhoneNumberCommand(output, context);
    };
    return OptInPhoneNumberCommand;
}($Command));
export { OptInPhoneNumberCommand };
