// @aws-sdk/client-sns/dist/es/commands/SetEndpointAttributesCommand.js
import { __extends } from "tslib";
import { SetEndpointAttributesInput } from "../models/models_0";
import { deserializeAws_querySetEndpointAttributesCommand, serializeAws_querySetEndpointAttributesCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
var SetEndpointAttributesCommand = /** @class */ (function (_super) {
    __extends(SetEndpointAttributesCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function SetEndpointAttributesCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    SetEndpointAttributesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SNSClient";
        var commandName = "SetEndpointAttributesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SetEndpointAttributesInput.filterSensitiveLog,
            outputFilterSensitiveLog: function (output) { return output; },
        };
        if (typeof logger.info === "function") {
            logger.info({
                clientName: clientName,
                commandName: commandName,
            });
        }
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SetEndpointAttributesCommand.prototype.serialize = function (input, context) {
        return serializeAws_querySetEndpointAttributesCommand(input, context);
    };
    SetEndpointAttributesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_querySetEndpointAttributesCommand(output, context);
    };
    return SetEndpointAttributesCommand;
}($Command));
export { SetEndpointAttributesCommand };
