// @aws-sdk/client-sns/dist/es/commands/SetSMSAttributesCommand.js
import { __extends } from "tslib";
import { SetSMSAttributesInput, SetSMSAttributesResponse } from "../models/models_0";
import { deserializeAws_querySetSMSAttributesCommand, serializeAws_querySetSMSAttributesCommand, } from "../protocols/Aws_query";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
var SetSMSAttributesCommand = /** @class */ (function (_super) {
    __extends(SetSMSAttributesCommand, _super);
    // Start section: command_properties
    // End section: command_properties
    function SetSMSAttributesCommand(input) {
        var _this = 
        // Start section: command_constructor
        _super.call(this) || this;
        _this.input = input;
        return _this;
        // End section: command_constructor
    }
    SetSMSAttributesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SNSClient";
        var commandName = "SetSMSAttributesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: SetSMSAttributesInput.filterSensitiveLog,
            outputFilterSensitiveLog: SetSMSAttributesResponse.filterSensitiveLog,
        };
        if (typeof logger.info === "function") {
            logger.info({
                clientName: clientName,
                commandName: commandName,
            });
        }
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    SetSMSAttributesCommand.prototype.serialize = function (input, context) {
        return serializeAws_querySetSMSAttributesCommand(input, context);
    };
    SetSMSAttributesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_querySetSMSAttributesCommand(output, context);
    };
    return SetSMSAttributesCommand;
}($Command));
export { SetSMSAttributesCommand };
